nixos/modules/services/networking/keepalived/default.nix

```
@@ -282,6 +290,15 @@ in
 
     assertions = flatten (map vrrpInstanceAssertions vrrpInstances);
 
+    networking.firewall.extraCommands = lib.mkIf cfg.openFirewall ''
+      # Allow VRRP and AH packets
+      iptables -A nixos-fw -p vrrp -j ACCEPT
+      iptables -A nixos-fw -p ah -j ACCEPT
+
+      ip6tables -A nixos-fw -p vrrp -j ACCEPT
+      ip6tables -A nixos-fw -p ah -j ACCEPT
```

```suggestion
      ip46tables -A nixos-fw -p vrrp -m comment --comment "services.keepalived.openFirewall" -j ACCEPT
      ip46tables -A nixos-fw -p ah -m comment --comment "services.keepalived.openFirewall" -j ACCEPT
```

We have a `ip46tables` wrapper, that runs both `iptables` and `ip6tables`.

Additionally, I think a comment via `--comment` is more helpful here.
`services.prometheus.exporters.<name>.openFirewall` does the same[^1].

[^1]: https://github.com/NixOS/nixpkgs/blob/1382337b866618a7914d36afbc5014fc66404574/nixos/modules/services/monitoring/prometheus/exporters.nix#L226

---

Thank you for the feedback! Fixed here 739da3c